### PR TITLE
new(driverkit): allow configuring the module driver and device name

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

/area cmd

/area pkg

> /area docs

> /area tests

**What this PR does / why we need it**:

The falco libs client might require a kernel module name which is not called `falco` / `falco`, depending on the client. Allow these two options to be passed as command line flags via `--moduledrivername` and `--moduledevicename`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new(driverkit): add --moduledrivername and --moduledevicename to change the name of the built kernel module.
```

### CODE REVIEW

File: pkg/driverbuilder/builder/builders.go [hunk 1]
```diff
@@ -12,16 +12,16 @@ import (
 const DriverDirectory = "/tmp/driver"
 
 // ModuleFileName is the standard file name for the kernel module.
-const ModuleFileName = "falco.ko"
+const ModuleFileName = "module.ko"
```
Comment (leodido) [inline pkg/driverbuilder/builder/builders.go:15]: I'm not getting why this change, Luca.

Not that it is a huge problem, the patch looks good!

**Review by leodido — COMMENTED** [review]
Please update the README accordingly, too

Ty!

File: pkg/driverbuilder/builder/builders.go (hunk 1 above)
Comment (LucaGuerra) [reply pkg/driverbuilder/builder/builders.go:15]: It isn't really necessary, but I wanted to catch cases where the `falco` string is hardcoded so when running the script in the container it wouldn't "accidentally" work. Also, it's symmetrical with `probe.o`. No big deal either way though, let me know if you feel like it's better to leave it `falco` and I'll revert this change.

**Comment by LucaGuerra** [comment]
Thanks a lot for the review @leodido ! Good catch on the readme, I added a commit for the readme.

File: pkg/driverbuilder/builder/builders.go (hunk 1 above)
Comment (leodido) [reply pkg/driverbuilder/builder/builders.go:15]: Naaa, it makes sense. I understood your reasoning now.

File: cmd/root_options.go [hunk 2]
```diff
@@ -21,6 +21,8 @@ type RootOptions struct {
 	Architecture     string `default:"x86_64" validate:"required,oneof=x86_64" name:"architecture"`
 	DriverVersion    string `default:"dev" validate:"eq=dev|sha1|semver" name:"driver version"`
 	KernelVersion    uint16 `default:"1" validate:"omitempty,number" name:"kernel version"`
+	ModuleDriverName string `default:"falco" name:"kernel module driver name"`
```
Comment (leodido) [inline cmd/root_options.go:24]: Worth adding a validator to restrict the alphabet a driver/device name can have?

Not sure any character is possible...

**Review by leodido — COMMENTED** [review]
Left another comment :)

File: cmd/root_options.go (hunk 2 above)
Comment (LucaGuerra) [reply cmd/root_options.go]: Sorry about the delay. Anyways I went down that specific rabbit hole and it looks to me that:
* A kernel driver does not seem to have specific name restriction if not a max length of 60 characters ( https://elixir.bootlin.com/linux/latest/source/include/linux/moduleparam.h#L21 ). In fact, I tried editing the elf and changing it to `falco🦅` and it does work, although I wouldn't recommend it.
* The device is just a file name, so max 255 and no slash should be enough

I added a commit with these rules

Comment (leodido) [reply cmd/root_options.go]: Worth checking, innit?

**Review by leodido — APPROVED** [review]
